### PR TITLE
fix build on linux

### PR DESCRIPTION
**Pull request by jxo**

### CODE REVIEW

File: src/MernelPlatform/ByteOrderStream.hpp [hunk 1]
```diff
@@ -11,6 +11,16 @@
 #include <deque>
 #include <map>
 #include <type_traits>
+#include <limits>
+
+/// Conflicts with "endian.h" in unix
+#ifdef BIG_ENDIAN
```
Comment (mapron) [inline src/MernelPlatform/ByteOrderStream.hpp]: No, that's not good, I'd better rename that field insted. wait a minute.

Comment (mapron) [reply src/MernelPlatform/ByteOrderStream.hpp]: Done. Can you rebase please?

**Review by mapron — CHANGES_REQUESTED** [review]
Ok, just need rebase

File: src/MernelPlatform/ByteOrderStream.hpp (hunk 1 above)
Comment (jxo) [reply src/MernelPlatform/ByteOrderStream.hpp]: Yeah, rename is a better choice.

**Comment by mapron** [comment]
If it's unclear. you need to pull proper fix I made in my repo master to your fork version - master/main.
then rebase. then... well, github will show updated version of it.
Make sure to check build again when you remove unnessesary code (ifdef BIG_ENDIAN / cmake INCLUDES)

**Comment by mapron** [comment]
I will pull submodule after merge to FreeHereoes. Don't worry.

**Comment by jxo** [comment]
Yeah, I rebased and push force.
Is there a more convenient way to change submodule in github?

**Comment by mapron** [comment]
>  Is there a more convenient way to change submodule in github?

Dunno. Submodule thing overall is not about convenience.

**Review by mapron — APPROVED** [review]

**Pull request merged**